geomesa-blobstore/geomesa-blobstore-core/src/main/java/org/locationtech/geomesa/blob/core/impl/AccumuloGeoMesaBlobStore.java

```diff
@@ -0,0 +1,141 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.blob.core.impl;
+
+import org.geotools.data.Query;
+import org.locationtech.geomesa.accumulo.data.AccumuloDataStore;
+import org.locationtech.geomesa.accumulo.data.AccumuloDataStoreFactory;
+import org.locationtech.geomesa.accumulo.data.AccumuloDataStoreParams;
+import org.locationtech.geomesa.blob.core.AccumuloBlobStore;
+import org.locationtech.geomesa.blob.core.GeoMesaBlobStore;
+import org.opengis.filter.Filter;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class AccumuloGeoMesaBlobStore implements GeoMesaBlobStore {
+
+    protected final AccumuloBlobStore accumuloBlobStore;
+
+    public AccumuloGeoMesaBlobStore(final Map<String, Serializable> dataStoreParams) throws IOException {
+        final AccumuloDataStore ds = genNewADS(dataStoreParams);
+        if (ds == null) {
+            throw new IOException("Error initializing AccumuloGeoMesaBlobStore");
+        } else {
+            accumuloBlobStore = new AccumuloBlobStore(ds);
+        }
+    }
+
+    public AccumuloGeoMesaBlobStore(final String instanceId,
+                                    final String tableName,
+                                    final String zookeepers,
+                                    final String user,
+                                    final String password,
+                                    final String auths,
+                                    final Boolean useMock) throws IOException {
+        this(new HashMap<String, Serializable>() {{
+            put(AccumuloDataStoreParams.instanceIdParam().key, instanceId);
+            put(AccumuloDataStoreParams.tableNameParam().key, tableName);
+            put(AccumuloDataStoreParams.zookeepersParam().key, zookeepers);
+            put(AccumuloDataStoreParams.userParam().key, user);
+            put(AccumuloDataStoreParams.passwordParam().key, password);
+            put(AccumuloDataStoreParams.authsParam().key, auths);
+            put(AccumuloDataStoreParams.mockParam().key, useMock.toString());
```

Comment:
hehehe yeah i prefer to call the constructor from the other constructor usually in Java so I went for it.